### DISCUSSION TITLE
Execute Script/Programs

### DISCUSSION THREAD

**Discussion by gobo-ws** [diyhue/diyHue#1026] [Ideas]
Hello,
I used the Hue emulator [ha-bridge](https://github.com/bwssytems/ha-bridge/) from bwssystems a few years ago. However, it no longer works as intended so I am looking at diyHue.
Would it be possible to add a function to execute scripts/programs, for instance to call curl?

I used HA-Bridge, together with curl and [OLA](https://github.com/OpenLightingProject/ola), to control DMX devices with the official Hue app. Here’s my old how-to:

Add a new device in HA-Bridge.
Use the option: Execute Script/Program as Device Type (Informational)

OnUrl:
[{"item":"curl -d u=0 -d d=255 http://olahostname:9090/set_dmx","type":"cmdDevice"}]

DimUrl:
[{"item":"curl -d u=0 -d d=${intensity.math(X*1)} http://olahostname:9090/set_dmx","type":"cmdDevice"}]

OffUrl:
[{"item":"curl -d u=0 -d d=0 http://olahostname:9090/set_dmx","type":"cmdDevice"}]

Thanks in advance.

**Comment by mariusmotea** [comment]
Hi!

You can create a new protocol file and have the [tradfri](https://github.com/diyhue/diyHue/blob/master/BridgeEmulator/lights/protocols/tradfri.py) integration as example. For tradfri we execute coap-client-gnutls command using python subprocess library witch is similar with curl but for coap protocol.
